### DISCUSSION TITLE
Turbo Repo React package error on using useEffect

### DISCUSSION THREAD

**Discussion by roshanjayathunga** [vercel/turborepo#5164] [Help] [answered]
I have created react ui package using webpack and babel. It works properly without useState hook in apps folder after installing it as a package to one of my apps.

======
Following is the package.json file of that package.

{
  "name": "ui-presentation",
  "version": "1.0.0",
  "description": "",
  "module": "dist/components.bundle.min.js",
  "scripts": {
    "build": "webpack",
    "build-watch": "webpack --watch",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "RJ",
  "license": "UNLICENSED",
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "@babel/core": "^7.21.8",
    "@babel/preset-env": "^7.21.5",
    "@babel/preset-react": "^7.18.6",
    "babel-loader": "^9.1.2",
    "webpack-cli": "^5.1.1"
  }
}

========
Following is my package index.js file:
import React, {useEffect} from "react"

const Presentation = ({ presentation}) => {

    useEffect(() => {
      console.log("DDD")
    }, [])
    
    <div>
        Hello
    </div>
  
      );
}

export {Presentation} ;

=====

Why I am getting following error when trying to use this package in apps.

**components.bundle.min.js:2 Uncaught TypeError: Cannot read properties of null (reading 'useEffect')**

**Comment by DanielOberlechner** [comment]
How about creating a npx create-new-react-application which works out of the box and go from there?
Would that be maybe a solution? This above is pseudo code ...

Kind regards
Daniel

**Reply by roshanjayathunga** [reply]
Hi Daniel,
Thank you very much for your reply.
Initially, I thought in that way. But, I felt like this method would be easy for creating packages. 
Actually, one of the main applications in the apps directory was created by using Create-React-App.

I will try to create the same package using  Create-React-App and update the result here.

Thanks again!!!

**Comment by roshanjayathunga** [comment, marked as answer]
Hi Daniel,
Happy News! It works for me with CRA.
I found an good article here.
https://medium.com/hackernoon/creating-a-library-of-react-components-using-create-react-app-without-ejecting-d182df690c6b

Thank you very much for your idea!

I keep this discussion will anyone has some solution for Babel and Webpack.

**Reply by anthonyshew** [reply]
Thanks for providing your solution!
